ddl/ddl_api.go

```diff
@@ -2938,6 +2940,56 @@ func (d *ddl) ChangeColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Al
 	return errors.Trace(err)
 }
 
+// RenameColumn renames an existing column.
+func (d *ddl) RenameColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.AlterTableSpec) error {
+	oldColName := spec.OldColumnName.Name
+	newColName := spec.NewColumnName.Name
+	if oldColName.L == newColName.L {
+		return nil
+	}
+
+	is := d.infoHandle.Get()
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return errors.Trace(infoschema.ErrDatabaseNotExists)
+	}
+
+	tbl, err := is.TableByName(ident.Schema, ident.Name)
+	if err != nil {
+		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ident.Schema, ident.Name))
+	}
+
+	oldCol := table.FindCol(tbl.Cols(), oldColName.L)
```

Comment:
```suggestion
	oldCol := table.FindCol(tbl.VisibleCols(), oldColName.L)
```

Reply:
How to rename a hidden column then?

Reply:
@djshow832 We can't rename a hidden column